File: Simple Bank System/Simple Bank System.cpp
// Solution for Simple Bank System in CPP

class Bank {
public:

    unordered_map<int, long long> accToBalance;

    Bank(vector<long long>& balance) {
        for (int i = 0; i < balance.size(); i++) {
            accToBalance[i + 1] = balance[i];
        }
    }
    
    bool transfer(int account1, int account2, long long money) {
        if (!accToBalance.count(account1) || !accToBalance.count(account2) || 
            accToBalance[account1] < money) {
            return false;
        }
        accToBalance[account1] -= money;
        accToBalance[account2] += money;
        return true;
    }
    
    bool deposit(int account, long long money) {
        if (!accToBalance.count(account)) {
            return false;
        }
        accToBalance[account] += money;
        return true;
    }
    
    bool withdraw(int account, long long money) {
        if (!accToBalance.count(account) || accToBalance[account] < money) {
            return false;
        }
        accToBalance[account] -= money;
        return true;
    }
};

/**
 * Your Bank object will be instantiated and called as such:
 * Bank* obj = new Bank(balance);
 * bool param_1 = obj->transfer(account1,account2,money);
 * bool param_2 = obj->deposit(account,money);
 * bool param_3 = obj->withdraw(account,money);
 */

File: Simple Bank System/Simple Bank System.py
# Solution for Simple Bank System in PY

class Bank:
    def __init__(self, balance: List[int]):
        self.balance = balance

    def transfer(self, account1: int, account2: int, money: int) -> bool:
        if not self._valid(account1) or not self._valid(account2) or self.balance[account1 - 1] < money:
            return False
        self.balance[account1 - 1] -= money
        self.balance[account2 - 1] += money
        return True

    def deposit(self, account: int, money: int) -> bool:
        if not self._valid(account):
            return False
        self.balance[account - 1] += money
        return True

    def withdraw(self, account: int, money: int) -> bool:
        if not self._valid(account) or self.balance[account - 1] < money:
            return False
        self.balance[account - 1] -= money
        return True

    def _valid(self, account: int) -> bool:
        return 1 <= account <= len(self.balance)


# Example usage:
# obj = Bank([10, 100, 20, 50, 30])
# print(obj.transfer(2, 3, 50))
# print(obj.deposit(1, 20))
# print(obj.withdraw(3, 10))
